File: python/443.string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        i,j,n = 0,0,len(chars)
        prev = chars[0]
        cont = 0
        while j < n:
            if chars[j] != prev:
                chars[i] = prev
                i+=1
                if cont > 1:
                    l = 0
                    str_cont = str(cont)
                    while l < len(str_cont):
                        chars[i] = str_cont[l]
                        i+=1
                        l+=1
                cont = 0
            prev = chars[j]
            j+=1
            cont+=1
        if cont:
            l = 0
            chars[i] = prev
            i+=1
            if cont > 1:
             str_cont = str(cont)
             while l < len(str_cont):
                 chars[i] = str_cont[l]
                 i+=1
                 l+=1
        return i

File: cpp/443.string-compression.cpp
class Solution {
public:
    int solve(vector<char> &chars){
	int i = 0, j = 0, n = chars.size(), cont=0;
	char prev = chars[0];
	while(j < n){
		if(chars[j] != prev){
			auto str_cont = to_string(cont);
			chars[i++] = prev;
			for(int k = 0 ; cont>1 && k < str_cont.size(); k++){
				chars[i++] = str_cont[k];
			}
			cont = 0;
		}
		cont++;
		prev = chars[j++];
	}
	if(cont){
	   auto str_cont = to_string(cont);
	   chars[i++] = prev;
	    for(int k = 0 ; cont >1 && k < str_cont.size(); k++)chars[i++] = str_cont[k];
	}
	return i;
    }
    int compress(vector<char>& chars) {
	    return solve(chars);
    }
};
